### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by hshaosf**
Initial commit for middleware.py

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# SFDS middleware.py
+SFDS middleware.py jumpstarts your next python-based middleware microservice. It consist of a skeleton boilerplate make up of
```
Comment (akegan) [inline README.md]: typo consist -> consists

File: README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+* [falcon](https://falconframework.org/): bare-metal Python web API framework 
+* [gunicorn](https://gunicorn.org/): Python WSGI HTTP Server for UNIX
+* [pytest](https://docs.pytest.org/en/latest/): Python testing tool 
+* [sentry](https://sentry.io/): error tracking tool
+* [jsend](https://github.com/omniti-labs/jsend):  a specification for a simple, no-frills, JSON based format for application-level communication
+
+## Requirement
+* Python3 
+([Mac OS X](https://docs.python-guide.org/starting/install3/osx/) / [Windows](https://www.stuartellis.name/articles/python-development-windows/)
+* Pipenv & Virtual Environments ([virtualenv](https://docs.python-guide.org/dev/virtualenvs/#virtualenvironments-ref) / [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/))
```
Comment (akegan) [inline README.md:13]: did we think about using docker for env for middleware? Virtualenv is great until you occasionally everything somehow goes horribly wrong

**Review by akegan — COMMENTED** [review]
looks good to me so far! can we change the repo name to be something that's easier to remember though? Like python-middleware-template, or something like that? Starting with sfds seems redundant and I had a hard time finding the PR because I couldn't remember the repo name

File: README.md (hunk 2 above)
Comment (hshaosf) [reply README.md:13]: I thought part of virtualenv is creating isolated environments so things don't go horribly wrong.

**Comment by hshaosf** [comment]
I think I going to create new PR to rename the project.  Sasha mentioned we should call this microservice instead of middleware.

File: README.md (hunk 2 above)
Comment (akegan) [reply README.md:13]: it is! it just goes horribly wrong sometimes too, more often than docker goes horribly wrong (which also sometimes happens!)

Comment (hshaosf) [reply README.md:13]: we can make docker a feature request. :)

**Review by akegan — DISMISSED** [review]

**Comment by akegan** [comment]
@hshaosf I've approved, but it doesn't looks like it counts because I don't have write access? Can you add me to the repo?

Please fix the typo in the readme before merge though.

File: README.md (hunk 1 above)
Comment (hshaosf) [reply README.md]: fixed via b31f836daf60cdcc070f70a55caf3fdf4cf9bb6f

**Comment by hshaosf** [comment]
@akegan permission updated. typo fixed checked in.  would need to approval again. 👍

**Review by akegan — APPROVED** [review]

**Pull request merged**